xmldb/database.go

```diff
@@ -0,0 +1,121 @@
+package xmldb
+
+import (
+	"errors"
+	"fmt"
+	"github.com/Project-Planner/backend/model"
+	"github.com/google/uuid"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strings"
+)
+
+// The struct implementing the web.Database interface
+type Database struct {
+	config DBConfig
+	auth   map[string]model.Login
+	users  map[string]model.User
+}
+
+func NewDatabase(config DBConfig) (Database, error) {
+	// 1. Step: Ensuring that the parent folders (auth, user, calendars) exist.
+	ensureDir(config.RootDir)
+	ensureDir(config.AuthDir)
+	ensureDir(config.UserDir)
+	ensureDir(config.CalendarDir)
+
+	//2. Step: Parsing source files into corresponding structs or struct collections.
+	// #Auth#
+	// a) Ensuring that the authentication file actually exists.
+	//	  If it doesn't, then it is created, so that parsing
+	//	  can go on.
+	var authPath = fmt.Sprintf("%s/%s", config.AuthDir, "/auth.xml")
+	var authPayload model.Auth
+	var auth = make(map[string]model.Login)
+	if !exists(authPath) {
+		create(authPath, authPayload.ToString())
+		log.Print("Authentication file didn't exist - was created now.")
+	}
+
+	// b) Parsing authentication file into struct.
+	parse(authPath, &authPayload)
+	for _, login := range authPayload.Logins {
+		auth[login.Name.Value] = login
+	}
+
+	// #Users#
+	// Users are each represented by a separate file inside the corresponding folder.
+	// Each file to be iterated over in this folder yields the users unique id (UUID)
+	// and content for the user struct to be parsed with.
+	var users = make(map[string]model.User)
+	filepath.Walk(config.UserDir, func(path string, info os.FileInfo, err error) error {
+
+		//For completeness, it is tested, whether the file
+		//has an "xml" file extension and hence can be seen as such.
+		var name = info.Name()
+		var matching, _ = regexp.MatchString(".+\\.xml$", name)
```

Comment:
In go kann man strings auch mit backticks `  angeben und dann werden sie wirklich buchstäblich (auch mit newline etc genommen). Nur so als Info am Rande. Das könnte dann so aussehen: 
```suggestion
		var matching, _ = regexp.MatchString(`.+\.xml$`, name)
```